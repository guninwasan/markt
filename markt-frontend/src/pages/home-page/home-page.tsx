import { ExploreListings, Navbar } from "../../components";
import { SearchBar } from "../../components/searchbar/searchbar";

const HomePage = () => {
  return (
    <div>
      <Navbar />
      <SearchBar></SearchBar>
      <img src="/" alt="image" width={"100%"} height={400} />
      <ExploreListings />
    </div>
  );
};
export { HomePage };
