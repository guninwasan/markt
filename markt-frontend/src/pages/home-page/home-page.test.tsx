import React from "react";
import { render, screen } from "@testing-library/react";
import { HomePage } from "./home-page";

jest.mock("../../components", () => ({
  ExploreListings: () => (
    <div data-testid="explore-listings">Explore Listings</div>
  ),
  Navbar: () => <div data-testid="navbar">Navbar</div>,
}));

describe("HomePage", () => {
  it("should render Navbar", () => {
    render(<HomePage />);
    const navbarElement = screen.getByTestId("navbar");
    expect(navbarElement).toBeInTheDocument();
  });

  it("should render the image", () => {
    render(<HomePage />);
    const imageElement = screen.getByAltText("Cover Image");
    expect(imageElement).toBeInTheDocument();
  });

  it("should render Explore Listings component", () => {
    render(<HomePage />);
    const exploreListingsElement = screen.getByTestId("explore-listings");
    expect(exploreListingsElement).toBeInTheDocument();
  });

  it("should match the snapshot", () => {
    const { container } = render(<HomePage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
