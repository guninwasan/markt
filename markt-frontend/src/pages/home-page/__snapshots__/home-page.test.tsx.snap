// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match the snapshot 1`] = `
<div>
  <div
    data-testid="navbar"
  >
    Navbar
  </div>
  <img
    alt="Cover Image"
    src="/cover-image3.jpg"
    style="width: 100%; height: 400px; object-fit: cover; object-position: bottom;"
  />
  <div
    data-testid="explore-listings"
  >
    Explore Listings
  </div>
</div>
`;
